js/blocks/components/repeater-rows.js

```diff
@@ -129,26 +129,7 @@ import { Fields } from './';
 			 * Without this, it looks like setAttributes() doesn't recognize a change to the array, and the component doesn't re-render.
 			 */
 			const rows = repeaterRows.slice();
-
-			/*
-			 * Ensure that every row has the required attributes, so that we don't lose blank rows.
-			 */
-			for ( name in this.props.subFields ) {
-				rows.forEach( ( row ) => {
-					if ( ! row.hasOwnProperty( name ) ) {
-						row[ name ] = null;
-					}
-				} );
-			};
-
-			rows.splice(
-				to,
-				0,
-				rows.splice(
-					from,
-					1
-				)[0]
-			);
+			[ rows[ from ], rows[ to ] ] = [ rows[ to ], rows[ from ] ]
```

Comment:
Wow – this is so so obvious. Can't believe how much I over-engineered this! Pickup of the week!

Reply:
Haha, I didn't know about that either. I just happened upon it from StackOverflow.

Reply:
The previous solution actually worked fine.